### ISSUE TITLE
Search is incredibly slow (increasingly maybe?)

### ISSUE THREAD

**Issue by jonolds** [microsoft/vcpkg#42680]
vcpkg search is taking almost 2 minutes to complete each time. I tried disabling windows firewall and there was no change.

**Environment**
windows 11 latest

**To Reproduce**
vcpkg search package_name 
....wait...and wait....

**Expected behavior**
results within a few seconds

**Failure logs**
are there logs to check for this?

**Comment by FrankXie05** [comment]
@jonolds You can try to search for more precise keywords. The search function of vcpkg will display all the information about your search (including dependencies and features).

**Comment by JackBoosY** [comment]
Add extra option `--debug`, try search again and provide the output here?

**Comment by dg0yt** [comment]
Almost all time spent in filesystem?
Is the filesystem or drive healthy?

**Comment by jonolds** [comment]
Yes, (or I believe so), it's pretty snappy overall, but I'm not sure how to demonstrate that. Building OpenCV with contrib and cuda (and using vcpkg for a lot of dependencies) takes 20-25 mins. 
this is from chkdsk:
```
Windows has scanned the file system and found no problems.
No further action is required.

 976269311 KB total disk space.
 646462456 KB in 1808575 files.
   1046856 KB in 385816 indexes.
         0 KB in bad sectors.
   3449507 KB in use by the system.
     65536 KB occupied by the log file.
 325310492 KB available on disk.

      4096 bytes in each allocation unit.
 244067327 total allocation units on disk.
  81327623 allocation units available on disk.
Total duration: 2.70 minutes (162082 ms).
```

Yes on the time spent in the filesystem, from what I can tell. Everything before the packages list happens within a few seconds. 

Is the ports directory the only thing searched?
Also, I just noticed this line:

```
[DEBUG] D:\a\_work\1\s\src\vcpkg\commands.find.cpp(208)
```

Do you know what that line of output is for? I have an external drive D: that's totally unrelated, so unless that's something expected that might be related. Thank you!

**Comment by JackBoosY** [comment]
> Yes, (or I believe so), it's pretty snappy overall, but I'm not sure how to demonstrate that. Building OpenCV with contrib and cuda (and using vcpkg for a lot of dependencies) takes 20-25 mins. this is from chkdsk:
> 
> ```
> Windows has scanned the file system and found no problems.
> No further action is required.
> 
>  976269311 KB total disk space.
>  646462456 KB in 1808575 files.
>    1046856 KB in 385816 indexes.
>          0 KB in bad sectors.
>    3449507 KB in use by the system.
>      65536 KB occupied by the log file.
>  325310492 KB available on disk.
> 
>       4096 bytes in each allocation unit.
>  244067327 total allocation units on disk.
>   81327623 allocation units available on disk.
> Total duration: 2.70 minutes (162082 ms).
> ```
> 
> Is the ports directory the only thing searched?

Correct, only ports folder can be searched.

>Also, I just noticed this line:
> 
> ```
> [DEBUG] D:\a\_work\1\s\src\vcpkg\commands.find.cpp(208)
> ```
> 
> Do you know what that line of output is for? I have an external drive D: that's totally unrelated, so unless that's something expected that might be related. Thank you!

It's not related to your machine. The path is in the machine which build the vcpkg binary release.

> ```
> C:\Users\WDAGUtilityAccount>cd c:\vcpkg
> 
> c:\vcpkg>vcpkg search bzip --debug
> [DEBUG] To include the environment variables in debug output, pass --debug-env
> [DEBUG] Trying to load bundleconfig from c:\vcpkg\vcpkg-bundle.json
> [DEBUG] Failed to open: c:\vcpkg\vcpkg-bundle.json
> [DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
> [DEBUG] Metrics enabled.
> [DEBUG] Feature flag 'binarycaching' unset
> [DEBUG] Feature flag 'compilertracking' unset
> [DEBUG] Feature flag 'registries' unset
> [DEBUG] Feature flag 'versions' unset
> [DEBUG] Feature flag 'dependencygraph' unset
> [DEBUG] Detected Container Execution Service
> [DEBUG] Using scripts-root: C:\vcpkg\scripts
> [DEBUG] Using builtin-ports: C:\vcpkg\ports
> [DEBUG] Using installed-root: C:\vcpkg\installed
> [DEBUG] Using buildtrees-root: C:\vcpkg\buildtrees
> [DEBUG] Using packages-root: C:\vcpkg\packages
> [DEBUG] Using vcpkg-root: C:\vcpkg
> [DEBUG] Using builtin-registry: C:\vcpkg\versions
> [DEBUG] Using downloads-root: C:\vcpkg\downloads
> [DEBUG] External asset downloads are allowed (x-block-origin is disabled)...
> [DEBUG] Asset cache is not configured.
> boost-iostreams[bzip2]                    Support bzip2 filters
> bzip2                    1.0.8#6          bzip2 is a freely available, patent free, high-quality data compressor. It...
> bzip2[tool]                               Builds bzip2 executable
> coin[bzip2]                               Support bzip2 compressed fonts
> ffmpeg[bzip2]                             Bzip2 support
> folly[bzip2]                              Support bzip2 for compression
> freetype[bzip2]                           Support bzip2 compressed fonts.
> gstreamer[bzip2-bad]                      Enable bzip2 stream compression in bad plugins
> gstreamer[bzip2-good]                     Enable bzip2 stream compression in good plugins
> hpx[bzip2]                                Build with bzip2 compression
> htscodecs[bzip2]                          Enable support for BZ2 within Arith streams
> htslib[bzip2]                             Enable support for BZ2-compressed CRAM files
> kf5archive[bzip2]                         Support for bzip2 compressed files and data streams
> libarchive[bzip2]                         BZip2 support
> libmagic[bzip2]                           Enable bzip2 support
> libsbml[bzip2]                            bz2 compression support for libsbml
> libzip                   1.10.1           A library for reading, creating, and modifying zip archives.
> libzip[bzip2]                             Support bzip2-compressed zip archives
> libzip[commoncrypto]                      AES (encryption) support using Apple's Common Crypto API
> libzip[default-aes]                       Use default AES
> libzip[liblzma]                           Support XZ compressed zip archives using liblzma
> libzip[mbedtls]                           AES (encryption) support using mbedtls
> libzip[openssl]                           AES (encryption) support using OpenSSL
> libzip[wincrypto]                         AES (encryption) support using Windows Crypto API
> libzip[zstd]                              Support Zstandard-compressed zip archives
> libzippp                 7.1-1.10.1       Simple basic C++ wrapper around the libzip library. It is meant to be a po...
> libzippp[encryption]                      Support encryption
> minizip[bzip2]                            Support compression using bzip2 library
> minizip-ng[bzip2]                         Enables BZIP2 compression
> quazip[bzip2]                             BZIP2 compression
> rocksdb[bzip2]                            build with bzip2
> The result may be outdated. Run `git pull` to get the latest results.
> If your port is not listed, please open an issue at and/or consider making a pull request.  -  https://github.com/Microsoft/vcpkg/issues
> [DEBUG] D:\a\_work\1\s\src\vcpkg\commands.find.cpp(208):
> [DEBUG] Time in subprocesses: 360611us
> [DEBUG] Time in parsing JSON: 36878us
> [DEBUG] Time in JSON reader: 34149us
> [DEBUG] Time in filesystem: 745576us
> [DEBUG] Time in loading ports: 932396us
> [DEBUG] Exiting after 1.3 s (919626us)
> ```

You can see the search command takes 1.3s, which should be normal.
It should be related to your VM if you are using VM.

**Comment by jonolds** [comment]
ohhh, that makes sense about the drive. thank you!

**Comment by BillyONeal** [comment]
Time appears consistent with the hardware in use. I checked and this is not a duplicate of https://github.com/microsoft/vcpkg/issues/31133 because manifest mode is not in use.
